src/components/rich-text-widget.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { getContentHeight, htmlToSlate, SlateEditor, slateToHtml, SlateToolbar } from "@concord-consortium/slate-editor";
+
+import "@concord-consortium/slate-editor/build/index.css";
+import "./rich-text-widget.css";
+
+const kThemeColor = "#34a5be";
+
+interface IRichTextProps {
+  id: string;
+  onBlur:  (id: string, value:string) => void;
+  value: string;
+}
+
+function exportHtml(value: any) {
+  const html = slateToHtml(value);
+  // convert empty paragraph to empty string
+  const emptyParagraphsRegex = /<p>\s*<\/p>/gi;
+  const cleanedHTML = html.replace(emptyParagraphsRegex, "");
+  return cleanedHTML;
+}
+
+export const RichTextWidget = (props: IRichTextProps) => {
+  const { id, onBlur } = props;
+  const [value, setValue] = useState(htmlToSlate(props.value || ""));
+  const [changeCount, setChangeCount] = useState(0);
+  const editorRef = useRef<any>();
+  const kExtraHeight = 30;
+  const kInitialHeight = 50;
+  const [height, setHeight] = useState(kInitialHeight);
+
+  const handleLoad = useCallback(() => {
```

Comment:
Just curious why this is a callback since it is wrapping a setter that itself uses a callback - `count` isn't captured.